content/docs/userspace/http-api-guide.md

```diff
@@ -0,0 +1,649 @@
+The same conversion will happen in reverse for responses.
+
+Note that Eyre makes a best effort attempt to convert data to and from JSON. If
+the marks in question do not contain appropriate JSON conversion functions, it
+will fail. Not all [scry endpoints](#scry-endpoints), [subscription
+paths](#subscriptions), pokes, etc, are intended to be used from a front-end, so
+not all use marks which can be converted to and from JSON (the `noun` mark for
+example). If documentation is available for the agent or thread in question, is
+should note in some fashion whether it can take or produce JSON. The majority of
+things you'll want to interact with through Eyre _will_ work with JSON.
+
+### Gall agents
+
+An _agent_ is a userspace application managed by the [Gall] vane. A desk may
+contain multiple agents that do different things. The Groups app, for example,
+has `graph-store`, `group-store`, `contact-store`, and others in its desk.
+Agents are the main thing you'll interact with through Eyre. They have a simple
+interface with three main parts:
+
+#### Pokes
+
+Pokes are single, stand-alone messages to agents. Pokes are how you send data
+and requests to agents. Agents will send back either a positive acknowledgement
+(ack) or a negative acknowledgement (nack). The agent can't send actual data
+back in the acks. If they have any response to give back, it will be sent out to
+subscribers on a subscription path instead.
+
+The pokes an agent accepts will be defined in the `on-poke` section of its
+source in the desk's `/app` directory, and typically in its type definition file
+in the `/sur` directory too. If the agent has documentation, the data structures
+and marks of its pokes should be listed there.
+
+`http-api` includes a `poke` function which allows you to perform pokes through
+Eyre, and is [detailed below](#poke).
+
+#### Subscriptions
+
+Agents define subscription paths which you can subscribe to through Eyre. A path
+might be simple and fixed like `/foo/bar`, or it might have variable elements
+where you can specify a date, ship, or what have you. Each agent will define its
+subscription paths in the `on-watch` section of its source, and if it has
+documentation the paths should also be listed there.
+
+You can subscribe by sending a watch request to the agent with the desired path
+specifed. The agent will apply some logic (such as checking permissions) to the
+request and then ack or nack it. If acked, you'll be subscribed and begin
+receiving any updates the agent sends out on that path. What you'll receive on a
+given path depends entirely on the agent.
+
+Agents can kick subscribers, and you can also unsubscribe at any time.
+
+`http-api` includes a `subscribe` function which allows you to perform pokes through
+Eyre, and is [detailed below](#subscribe).
+
+#### Scry Endpoints
+
+Pokes and subscriptions can modify the state of the agent. A third kind of
+interaction called a _scry_ does not - it simply retrieves data from the agent
```

Comment:
If the audience is not-particularly-Hoonish devs (is it?) then a little table here may be helpful.